tests/common.go

```diff
@@ -13948,6 +13949,76 @@ func GetVolumeNamefromPVC(namespace string) ([]string, error) {
 	return nil, fmt.Errorf("No PVCs found in namespace [%s]", namespace)
 }
 
+func DoVolumeQuorumValidation(volumeQuorumValidationStopSignal chan struct{}, vQourumError *error) {
+	itr := 1
+	for {
+		select {
+		case <-volumeQuorumValidationStopSignal:
+			log.Infof("Exiting volume quorum validation routine")
+			return
+		default:
+			log.Infof("Volume quorum validation iteration: #%d", itr)
+			errorChan := make(chan error, 50)
+			ValidateVolumeQuorum(&errorChan)
+			close(errorChan) // Close the channel to signal completion
+			for err := range errorChan {
+				if err != nil {
+					*vQourumError = multierr.Append(*vQourumError, err)
+				}
+			}
+			if *vQourumError != nil {
+				return
+			}
+			itr++
+			time.Sleep(10 * time.Second)
+		}
+	}
+}
+
+func ValidateVolumeQuorum(errChan ...*chan error) {
+	volIDs, err := Inst().V.ListAllVolumes()
+	if err != nil {
+		// add to errChan
+		err := fmt.Errorf("error listing volumes, Err: %v", err)
+		processError(err, errChan...)
+		return
+	}
+
+	for _, volID := range volIDs {
+		apiVol, err := Inst().V.InspectVolume(volID)
+		if err != nil {
+			err = fmt.Errorf("error inspecting volume [%s], Err: %v", volID, err)
+			processError(err, errChan...)
+			return
+		}
+		// check if volume is up
+		log.Infof("Volume [%s] status: %v", volID, apiVol.Status)
+		if apiVol.Status != opsapi.VolumeStatus_VOLUME_STATUS_UP {
+			// check if volume replicas are on different nodes
+			// get all the nodes where replicas are present
+			replicas := apiVol.ReplicaSets
+			if len(replicas) == 0 {
+				err := fmt.Errorf("volume [%s] does not have any replicas", volID)
+				processError(err, errChan...)
+				return
+			}
+
+			replicaNodes := replicas[0].Nodes
+
+			// if it is repl-2 volume and replicas are on same node, do not fail the test
+			// if it is repl-3 volume and replicas or either on same or 2 different nodes, do not fail the test
```

Comment:
If there is a scenario 2 , then it is a bug on how the volume replicas are present on same node. But IMO it is not a bug that should fail this scenario, because upgrades will happen with volume uptime only if replicas are available on different nodes. Please suggest.